### PR TITLE
Dockerfile: Cleanups and enable ARM/AArch64 builds

### PR DESCRIPTION
**Pull request by philmd**
Hi,

This series enable the building of EDK2 AArch64 images within your handy docker image.
Trivial cleanups around added.

Thanks,

Phil.

### CODE REVIEW

**Review by pietrushnic — APPROVED** [review]
@philmd Thank you very much for this contribution. Changes looks good to me. @miczyg1 we should think about releasing images in November cycle.

**Comment by philmd** [comment]
@miczyg1 ping?

**Comment by miczyg1** [comment]
@philmd sorry for the delay. I am struggling to find free cycles to test it. I will verify the change and release new image this week, hopefully.

**Comment by philmd** [comment]
@miczyg1 OK I understand. I updated the README so testing should be easier.
@pietrushnic I added the ARMhf target, do you mind reviewing it again?
Thanks both of you!

**Comment by miczyg1** [comment]
Great! I really appreciate that README update @philmd

**Review by pietrushnic — APPROVED** [review]
@philmd more cool stuff. Thank you.

File: README.md [hunk 1]
```diff
@@ -36,8 +36,43 @@ $ cd ../../../..
...
+$ git clone https://github.com/tianocore/edk2-platforms.git
+$ git clone -b OpenSSL_1_1_0-stable https://github.com/openssl/openssl edk2/CryptoPkg/Library/OpensslLib/openssl
+$ docker run --rm -it -w /home/edk2/edk2 -v $PWD/edk2:/home/edk2/edk2 \
+-v $PWD/edk2-platforms:/home/edk2/edk2-platforms \
+-v ${CCACHE_DIR:-$HOME/.ccache}:/home/edk2/.ccache \
+3mdeb/edk2 /bin/bash
+(docker)$ . edksetup.sh
+(docker)$ make -C BaseTools
+(docker)$ export PACKAGES_PATH=/home/edk2/edk2:/home/edk2/edk2-platforms
+(docker)$ GCC5_ARM_PREFIX=arm-linux-gnueabihf- build -a ARM -t GCC5 -p Platform/ARM/VExpressPkg/ArmVExpress-CTA15-A7.dsc
```
Comment (miczyg1) [inline README.md:82]: @philmd it looks like there is some problem during main volume generation:

```
Generating FVMAIN FV
#Return Value = 2
GenSec: ERROR 0001: Error opening file


  /home/edk2/edk2/Build/ArmVExpress-CTA15-A7/DEBUG_GCC5/ARM/Platform/ARM/VExpressPkg/Drivers/ArmVExpressDxe/ArmHwDxe/OUTPUT/ArmHwDxe.efi
GenSec: ERROR 2000: Status is not successful
GenFds.py...
 : error 7000: Failed to generate section
  Status value is 0x2
```

File: README.md [hunk 2]
```diff
@@ -46,3 +46,38 @@ $ docker run --rm -it -w /home/edk2 -v $PWD/edk2:/home/edk2/edk2 \
...
 ```
+
+Building firmware for QEMU AArch64
+----------------------------------
+
+```
+$ docker pull 3mdeb/edk2
+$ git clone https://github.com/tianocore/edk2.git
+$ git clone https://github.com/tianocore/edk2-platforms.git
+$ git clone -b OpenSSL_1_1_0-stable https://github.com/openssl/openssl edk2/CryptoPkg/Library/OpensslLib/openssl
```
Comment (krystian-hebel) [inline README.md:57]: `OpenSSL_1_1_0-stable` seems to be moved (so much for `stable`...). I suggest dropping this line and using git submodule instead, see next comment.

File: README.md [hunk 3]
```diff
@@ -46,3 +46,38 @@ $ docker run --rm -it -w /home/edk2 -v $PWD/edk2:/home/edk2/edk2 \
...
+$ docker run --rm -it -w /home/edk2/edk2 -v $PWD/edk2:/home/edk2/edk2 \
+-v $PWD/edk2-platforms:/home/edk2/edk2-platforms \
+-v ${CCACHE_DIR:-$HOME/.ccache}:/home/edk2/.ccache \
+3mdeb/edk2 /bin/bash
+(docker)$ . edksetup.sh
```
Comment (krystian-hebel) [inline README.md:62]:
```suggestion
(docker)$ git submodule update --init --checkout
(docker)$ . edksetup.sh
```

File: README.md [hunk 4]
```diff
@@ -46,3 +46,38 @@ $ docker run --rm -it -w /home/edk2 -v $PWD/edk2:/home/edk2/edk2 \
...
+```
+
+Building firmware for Versatile Express ARM
+-------------------------------------------
+
+```
+$ docker pull 3mdeb/edk2
+$ git clone https://github.com/tianocore/edk2.git
+$ git clone https://github.com/tianocore/edk2-platforms.git
+$ git clone -b OpenSSL_1_1_0-stable https://github.com/openssl/openssl edk2/CryptoPkg/Library/OpensslLib/openssl
```
Comment (krystian-hebel) [inline README.md:74]: As above

File: README.md (hunk 1 above)
Comment (krystian-hebel) [reply README.md:82]: @miczyg1 for me it fails earlier:
```
cp -f /home/edk2/edk2/Build/ArmVExpress-CTA15-A7/DEBUG_GCC5/ARM/edk2-platforms/Platform/ARM/VExpressPkg/Drivers/ArmVExpressFastBootDxe/ArmVExpressFastBootDxe/DEBUG/*.map /home/edk2/edk2/Build/ArmVExpress-CTA15-A7/DEBUG_GCC5/ARM/edk2-platforms/Platform/ARM/VExpressPkg/Drivers/ArmVExpressFastBootDxe/ArmVExpressFastBootDxe/OUTPUT
make: *** No rule to make target '/home/edk2/edk2/Build/ArmVExpress-CTA15-A7/DEBUG_GCC5/ARM/Platform/ARM/VExpressPkg/Drivers/ArmVExpressFastBootDxe/ArmVExpressFastBootDxe/OUTPUT/ArmVExpressFastBootDxe.map', needed by '/home/edk2/edk2/Build/ArmVExpress-CTA15-A7/DEBUG_GCC5/FV/Ffs/4004e454-89a0-11e3-89aa-97ef9d942abcArmVExpressFastBootDxe/4004e454-89a0-11e3-89aa-97ef9d942abc.map'.  Stop.


build.py...
 : error 7000: Failed to execute command
	make tbuild [/home/edk2/edk2/Build/ArmVExpress-CTA15-A7/DEBUG_GCC5/ARM/edk2-platforms/Platform/ARM/VExpressPkg/Drivers/ArmVExpressFastBootDxe/ArmVExpressFastBootDxe]


build.py...
 : error 7000: Failed to execute command
	make tbuild [/home/edk2/edk2/Build/ArmVExpress-CTA15-A7/DEBUG_GCC5/ARM/edk2-platforms/Platform/ARM/Drivers/BootMonFs/BootMonFs]


build.py...
 : error F002: Failed to build module
	/home/edk2/edk2-platforms/Platform/ARM/VExpressPkg/Drivers/ArmVExpressFastBootDxe/ArmVExpressFastBootDxe.inf [ARM, GCC5, DEBUG]

- Failed -
Build end time: 12:28:33, Mar.31 2020
Build total time: 00:00:51
```

Notice lack of `edk2-platforms` in first error, it was present in `cp` above. Everything works if `edk2-platforms` is a subdirectory of `edk2`, with `PACKAGES_PATH` changed accordingly.

**Review by krystian-hebel — APPROVED** [review]
@philmd with some small changes to the instructions (most likely because it was written more than a year ago and edk2 code changed since then) everything works. I'll merge it and update the README.

Sorry it took so long, this PR probably got lost somehow.

**Pull request merged**